cli_tools/common/utils/storage/storage_client.go

```diff
@@ -172,6 +184,27 @@ func (sc *Client) FindGcsFile(gcsDirectoryPath string, fileExtension string) (*s
 		"path %v doesn't contain a file with %v extension", gcsDirectoryPath, fileExtension)
 }
 
+func isDepthValid(lookupDepth int, lookupPath, objectPath string) bool {
```

Comment:
minor: The name isDepthValid took me a bit to understand. An option to consider is refactoring it to have a method such as `distance(gcsDirectoryPath, f string)` that returns the number of directories between gcsDirectoryPath and the file f. Eg:

distance('/tmp', '/tmp/file')  =>  0  (same directory)
distance('/', '/tmp/file')  =>  1  (one sub dir)

This formulation gives you a nice `if` statement at 173, and corresponds directly to your definitions of gcsDirectoryPath and lookupDepth.